conda/core/index.py

```diff
@@ -164,7 +164,6 @@ def __init__(
         self.expanded_channels = IndexedSet()
         for channel in self._channels:
             urls = Channel(channel).urls(True, subdirs)
-            check_allowlist(urls)
```

Comment:
@jezdez, I’ve never done this before. Can you please provide some tips? Should I add some new performance benchmarks?

Reply:
Just using the the `timeit` module in Python returned the following:

```python
In [25]: timeit(lambda: check_allowlist(["conda-forge", "defaults", "bioconda"]), number=100)
Out[25]: 0.002058250014670193
```

I don't think we have to worry about this function causing significant slow downs when place elsewhere in the code. Furthermore, the above benchmarks include run this code and we have not seen a significant slow down there.

Not sure what else to do. Please advise.

Reply:
Sorry, should have been more specific, the `timeit` module is what I would have proposed, but I'm not concerns about `check_allowlist` itself, but the runtime of the `Context.channels` property since it's used a lot

Reply:
We don't have a good performance specific test for codspeed to track that part, to compare it automatically, but if you could run a few manual tests with and without `check_allowlist` used (basically on main and this branch), that'd suffice

Reply:
@jezdez,

Here's the results of a few tests locally:

#### On bugfix branch

```python
In [1]: from conda.base.context import context

In [2]: context.__init__()

In [3]: from timeit import timeit

In [4]: timeit(lambda: context.channels, number=100)
Out[4]: 0.08723249999457039

In [5]: timeit(lambda: context.channels, number=100)
Out[5]: 0.006883459020173177

In [6]: timeit(lambda: context.channels, number=100)
Out[6]: 0.006838665984105319
```

#### On main branch

```python
In [1]: from conda.base.context import context

In [2]: context.__init__()

In [3]: from timeit import timeit

In [4]: timeit(lambda: context.channels, number=100)
Out[4]: 0.0017869579896796495

In [5]: timeit(lambda: context.channels, number=100)
Out[5]: 0.001080499991076067

In [6]: timeit(lambda: context.channels, number=100)
Out[6]: 0.0006820839771535248
```

You'll note the first run is longer on the bug fix branch. That's most likely a performance hit take to fill up the object cache for `Channel`. I'm not super worried about that because it was going to happen anyway during the execution on the program.

The slow down itself is measured in the thousandths of seconds, so I'm not sure if it's worth splitting hairs over to fix. This is also the difference in execution time over 100 calls.

I also experimented around with putting a `@cache` decorator on `check_allowlist`, and that gets us back to similar performance as before, but I'm not sure it's worth it. Caching a function call like that which is tied to global state could be super confusing in the future to debug 🙈. And, at that point, we might as we be caching the `Context.channels` property instead.

Reply:
You also have to remember the difference in perceived performance gains, especially when the user is using a channel that is not allowed. Right now, it would have to wait several seconds for the prefix data to get loaded, whereas now it fails immediately because none of that is run until that check passes.

Reply:
Right, FWIW the time is small, but the first run _is_ an order of magnitude slower, probably fine to ignore, thanks for doing the check